### DISCUSSION TITLE
Managing active sessions in users profile page

### DISCUSSION THREAD

**Discussion by mustaphaboudouch** [nextauthjs/next-auth#8147] [Help]
### Question 💬

I'm currently working on an application that requires managing active sessions for users on the profile page (store user devices in sessions table).

I couldn't find any existing implementation or documentation for this particular feature. I would greatly appreciate any guidance, code examples, or resources that could help me.

### How to reproduce ☕️

Here is an example of what I'm looking for :

![image](https://user-images.githubusercontent.com/64332287/236634242-095c5f8e-e33b-4204-8a3d-427e3bfda2ad.png)

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by dever23b** [comment]
I'm working on something related to terminating sessions.  Unfortunately, from what I have found, it's difficult at this time (see #7035).  

I've come up with a workaround that, while ugly and inefficient, seems to work for my purposes.  At the moment, there doesn't seem to be a way to interrupt the auth flow in NextAuth to prematurely terminate a session.  However, you can utilize the `jwt` and/or `session` callback to insert a flag that can instruct your front end to log the user out.  

Example (copied from my post in the other issue I referenced):
```typescript
// options.ts
const authOptions: NextAuthOptions = {
  //...
  callbacks: {
    session: async ({ session, token }) => {
      const state = authService.getState();

      if (state.isValid) {
        session.accessToken = state.accessToken;
        session.user = state.user;
      }

      if (state.mustReauthenticate) {
        session.terminate = true;
        session.expires = new Date().toDateString();
      }

      return session;
    },
  }
}
```

```typescript
// next-auth.d.ts
interface Session {
  user?: User;
  terminate?: boolean;
}
```

```typescript
// app/page.tsx
const Auth: React.FC = () => {
  const { data: session } = useSession();

  useEffect(() => {
    if (session?.terminate === true) {
      signOut();
    }
  }, [session]);

  return (
    <div>...</div>
  );
};
```

**Comment by stale** [comment]
It looks like this issue did not receive any activity for 60 days. It will be closed in 7 days if no further activity occurs. If you think your issue is still relevant, commenting will keep it open. Thanks!

**Comment by ThangHuuVu** [comment]
next-auth will create the session with this object type only:
```
{
  sessionToken: string
  userId: string
  expires: Date
}
```
so no information about the user-agent will be recorded at the session creation time. You will have to handle it in the app code - listen to the sign in status and record the user-agent information in the current session. 🙏
